polynote-frontend/polynote/ui.js

```diff
@@ -549,28 +549,32 @@ export class NotebookCellsUI extends UIEventTarget {
         window.addEventListener('resize', this.onWindowResize.bind(this));
     }
 
-    extractId(cellRef) {
-        if (typeof cellRef === 'string') {
-            return cellRef.match(/^Cell(\d+)$/)[1];
-        } else if (typeof cellRef === 'number') {
-            return cellRef;
-        } else {
-            throw { message: `Unable to parse cell reference ${cellRef}` };
-        }
+    firstCell() {
+        return this.getCells()[0];
+    }
+
+    getCell(cellId) {
+        return this.cells[cellId];
     }
 
-    getCell(cellRef) {
-        return this.cells[this.extractId(cellRef)];
+    getCells() {
+        return Array.from(this.el.children)
+            .map(container => container.cell)
+            .filter(cell => cell);
+    }
```

Comment:
This trick allows us to get the cells in order without needing to keep track of it explicitly